src/stream_parser/file_reader.rs

```diff
@@ -452,6 +471,32 @@ impl MaybeRepeatedType {
     }
 }
 
+#[derive(Debug)]
+struct FlagBits {
+    compat_flags: [u8; 8],
+    incompat_flags: [u8; 8],
+    appended_offsets: [u64; 3],
```

Comment:
What are appended_offsets for?

Reply:
It's to append data to a log when the flight controller crashes in the middle of writing.
Such logs will never be streamed to the suite.